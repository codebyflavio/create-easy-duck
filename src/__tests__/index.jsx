import { createTypes, createActionCreators, createReducer } from '../../dist'

test('should create base types', () => {
  expect(createTypes('RANDOM_NAME')).toMatchSnapshot()
  expect(createTypes('ANOTHER_NAME')).toMatchSnapshot()
})

test('should create action creators', () => {
  const action = createActionCreators('RANDOM_NAME')
  expect(action.start()).toMatchSnapshot()
  expect(action.finish({ hello: 'there' })).toMatchSnapshot()
  expect(action.error(new Error('An error occurred'))).toMatchSnapshot()
  expect(action.error()).toMatchSnapshot()
  expect(action.reset()).toMatchSnapshot()
})

test('should create reducer', () => {
  const TYPE = 'RANDOM_NAME'
  expect(createReducer(TYPE, { data: null })(undefined, {})).toMatchSnapshot()

  const action = createActionCreators(TYPE)
  const reducer = createReducer(TYPE)
  let state = reducer(undefined, {})
  expect(state).toMatchSnapshot()

  state = reducer(state, action.start())
  expect(state).toMatchSnapshot()

  state = reducer(state, action.finish({ it: 'works' }))
  expect(state).toMatchSnapshot()

  state = reducer(state, action.error())
  expect(state).toMatchSnapshot()

  state = reducer(state, action.reset())
  expect(state).toMatchSnapshot()
})
