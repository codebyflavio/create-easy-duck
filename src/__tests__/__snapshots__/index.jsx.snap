// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create action creators 1`] = `
Object {
  "type": "RANDOM_NAME/START",
}
`;

exports[`should create action creators 2`] = `
Object {
  "payload": Object {
    "hello": "there",
  },
  "type": "RANDOM_NAME/FINISH",
}
`;

exports[`should create action creators 3`] = `
Object {
  "error": "An error occurred",
  "type": "RANDOM_NAME/ERROR",
}
`;

exports[`should create action creators 4`] = `
Object {
  "error": "Something went wrong",
  "type": "RANDOM_NAME/ERROR",
}
`;

exports[`should create base types 1`] = `
Object {
  "ERROR": "RANDOM_NAME/ERROR",
  "FINISH": "RANDOM_NAME/FINISH",
  "RESET": "RANDOM_NAME/RESET",
  "START": "RANDOM_NAME/START",
}
`;

exports[`should create base types 2`] = `
Object {
  "ERROR": "ANOTHER_NAME/ERROR",
  "FINISH": "ANOTHER_NAME/FINISH",
  "RESET": "ANOTHER_NAME/RESET",
  "START": "ANOTHER_NAME/START",
}
`;

exports[`should create reducer 1`] = `
Object {
  "error": null,
  "payload": null,
  "status": Object {
    "isLoaded": false,
    "isLoading": false,
  },
}
`;

exports[`should create reducer 2`] = `
Object {
  "error": null,
  "payload": null,
  "status": Object {
    "isLoaded": false,
    "isLoading": false,
  },
}
`;

exports[`should create reducer 3`] = `
Object {
  "error": null,
  "payload": null,
  "status": Object {
    "isLoaded": false,
    "isLoading": true,
  },
}
`;

exports[`should create reducer 4`] = `
Object {
  "error": null,
  "payload": Object {
    "it": "works",
  },
  "status": Object {
    "isLoaded": true,
    "isLoading": false,
  },
}
`;

exports[`should create reducer 5`] = `
Object {
  "error": "Something went wrong",
  "payload": null,
  "status": Object {
    "isLoaded": true,
    "isLoading": false,
  },
}
`;

exports[`should create reducer 6`] = `
Object {
  "error": null,
  "payload": null,
  "status": Object {
    "isLoaded": false,
    "isLoading": false,
  },
}
`;
